File: src/examples/ruby/pickups/buy_pickup.rb
# frozen_string_literal: true

require 'date'
require 'easypost'
require 'dotenv'
require 'amazing_print'

Dotenv.load('/Users/jhammond/git/easypost/easypost-tools/.env')
EasyPost.api_key = ENV['EASYPOST_TEST_API_KEY']

pickup = EasyPost::Pickup.retrieve('pickup_123...')
pickup.buy(
  carrier: 'FedEx',
  service: 'Same-day Express Pickup'
)

ap pickup

File: src/examples/python/pickups/buy_pickup.py
import os

import easypost
from dotenv import load_dotenv

load_dotenv()
easypost.api_key = os.getenv('EASYPOST_TEST_API_KEY')

pickup = easypost.Pickup.retrieve("pickup_123")
pickup.buy(carrier="USPS", service="NextDay")

print(pickup)
